hack/istio/install-travel-agency-demo.sh

```diff
@@ -0,0 +1,261 @@
+#/bin/bash
+
+# This deploys the travel agency demo
+
+: ${CLIENT_EXE:=oc}
+: ${NAMESPACE_AGENCY:=travel-agency}
+: ${NAMESPACE_PORTAL:=travel-portal}
+: ${ENABLE_OPERATION_METRICS:=false}
+
+while [ $# -gt 0 ]; do
+  key="$1"
+  case $key in
+    -c|--client)
+      CLIENT_EXE="$2"
+      shift;shift
+      ;;
+    -eo|--enable-operation-metrics)
+      ENABLE_OPERATION_METRICS="$2"
+      shift;shift
+      ;;
+    -na|--namespace-agency)
+      NAMESPACE_AGENCY="$2"
+      shift;shift
+      ;;
+    -np|--namespace-portal)
+      NAMESPACE_PORTAL="$2"
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -c|--client: either 'oc' or 'kubectl'
+  -eo|--enable-operation-metrics: either 'true' or 'false'. Only works on Istio 1.6 installed in istio-system.
+  -na|--namespace-agency: where to install the travel agency demo resources
+  -np|--namespace-portal: where to install the travel portal demo resources
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+echo Will deploy Travel Agency using these settings:
+echo CLIENT_EXE=${CLIENT_EXE}
+echo NAMESPACE_AGENCY=${NAMESPACE_AGENCY}
+echo NAMESPACE_PORTAL=${NAMESPACE_PORTAL}
+echo ENABLE_OPERATION_METRICS=${ENABLE_OPERATION_METRICS}
+
+# Create the demo namespaces
+
+${CLIENT_EXE} create namespace ${NAMESPACE_AGENCY}
+${CLIENT_EXE} label namespace ${NAMESPACE_AGENCY} istio-injection=enabled
+
+${CLIENT_EXE} create namespace ${NAMESPACE_PORTAL}
+${CLIENT_EXE} label namespace ${NAMESPACE_PORTAL} istio-injection=enabled
+
+# Prepare the new demo namespaces for CNI
+
+if [ "${CLIENT_EXE}" == "oc" ]; then
+${CLIENT_EXE} adm policy add-scc-to-group privileged system:serviceaccounts:${NAMESPACE_AGENCY}
+${CLIENT_EXE} adm policy add-scc-to-group anyuid system:serviceaccounts:${NAMESPACE_AGENCY}
+cat <<EOF | ${CLIENT_EXE} -n ${NAMESPACE_AGENCY} create -f -
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+EOF
+
+${CLIENT_EXE} adm policy add-scc-to-group privileged system:serviceaccounts:${NAMESPACE_PORTAL}
+${CLIENT_EXE} adm policy add-scc-to-group anyuid system:serviceaccounts:${NAMESPACE_PORTAL}
+cat <<EOF | ${CLIENT_EXE} -n ${NAMESPACE_PORTAL} create -f -
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+EOF
+fi
+
+# Deploy the demo
+
+${CLIENT_EXE} apply -f <(curl -L https://raw.githubusercontent.com/lucasponce/travel-comparison-demo/master/travel_agency.yaml) -n ${NAMESPACE_AGENCY}
+${CLIENT_EXE} apply -f <(curl -L https://raw.githubusercontent.com/lucasponce/travel-comparison-demo/master/travel_portal.yaml) -n ${NAMESPACE_PORTAL}
```

Comment:
This is nice.
I wonder if we could add some parameter to also install the version 2 in an optional manner (Step [2.5.0]):
https://lucasponce.github.io/presentations/observability-in-action.html

Reply:
added that to the script